proposals/NNNN-vehicle-type-filter.md

```diff
@@ -0,0 +1,435 @@
+            <string>Ford</string>
+        </dict>
+        <dict>
+            <key>make</key>
+            <string>OEM1</string>
+        </dict>
+    </array>
+```
+#### Determine Vehicle type info
+
+1. After version negotiations and starting RPC service, the proxy should check if `Get vehicle type` is supported by the SDL enabled IVI system.
+2. If the protocol version supports `Get vehicle type`, the proxy will send the `Get vehicle type` protocol message.
+3. On receiving vehicle type information, proxy will check `SDLSupportedVehicleTypes` keys to check supported vehicle type.
+4. If the vehicle type is supported, the proxy should also notify the app about the connected vehicle type so that the app can configure SDL as required. The iOS proxy can implement `OnSDLEnabled` notification similar to that of an android proxy to notify the app about SDL connection with the supported vehicle. The app can provide life cycle configuration to the proxy on receiving `OnSDLEnabled` notification. 
+5. If the proxy determines from `Get vehicle type` ACK that the vehicle type is not supported, the proxy will end the RPC session. The application layer will not be notified about the vehicle type.
+6. If the protocol version does not support the `Get vehicle type` protocol message, the SDL proxy will continue with the app registration and it will rely on vehicle type information received in Register App interface response. If vehicle type is not supported, the exclusive apps will be allowed to unregister from SDL enabled system.
+7. If `Get vehicle type` message response is NACK, the proxy will end RPC session and the app will not be registered on the SDL system.
+
+![Sequence Diagram](../assets/proposals/NNNN-vehicle-type-filter/ios.png)
+
+## Potential downsides
+
+The Android SDL router service after transport connection needs to start RPC service. If the SDL router service does not start RPC service and relies on the client to send the `Get vehicle type` protocol message, the connected client would start another foreground service. This would force the connected app to show two notifications to users while determining vehicle type information and will need to terminate services if the app does not support SDL enabled IVI system. 
+
+## Impact on existing code
+
+Above mentioned changes need to be implemented in SDL Core, the android proxy and iOS proxy.
+
+## Alternatives considered
+
+### GetVehicleType as an RPC instead of Protocol message
+1. As per current implementation, the RAI is the first RPC sent to SDL enabled IVI system and the vehicle type information is available to the app in its response. 
+2. If `GetVehicleType` is an RPC, this behavior needs to be changed. The app should be able to use `GetVehicleType` RPC before `RegisterAppInterface` RPC.
+3. Since the RPC message version is sent to the app in `RAI response`, the app, when connected to older SDL enabled IVI, would not know if `GetVehicleType` RPC is supported or not.
```

Comment:
```suggestion
3. Since the RPC message version is sent to the app in `RAI response`, the app, when connected to an older SDL enabled IVI, would not know if the `GetVehicleType` RPC is supported or not.
```